### PR TITLE
THORN-2174 | Bump jaeger version to 0.30.6

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <ploffay@redhat.com>

https://issues.jboss.org/browse/THORN-2174

Changes:
* bump jaeger version to 0.31.0
* expose missing jaeger configuration to thorntail
* use default service name (`thorntail/unknown`) if it's not provided.

- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by pavolloffay** [comment]
@jpkrohling could you please have a look?

File: fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/JaegerFraction.java [hunk 1]
```diff
@@ -99,6 +113,8 @@ public String toString() {
                 ", agentPort='" + agentPort.get() + '\'' +
                 ", reporterFlushInterval='" + reporterFlushInterval.get() + '\'' +
                 ", reporterMaxQueueSize='" + reporterMaxQueueSize.get() + '\'' +
+                ", user='" + user.get() + '\'' +
+                ", password='" + password.get() + '\'' +
```
Comment (Ladicek) [inline fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/JaegerFraction.java]: I don't think that's a good idea.

File: fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java [hunk 2]
```diff
@@ -54,12 +58,11 @@ public void contextInitialized(ServletContextEvent servletContextEvent) {
 
         String serviceName = getProperty(sc, JAEGER_SERVICE_NAME);
         if (serviceName == null || serviceName.isEmpty()) {
-            logger.info("No Service Name set. Skipping initialization of the Jaeger Tracer.");
-            return;
+            logger.warn("No Service Name set. Using default. Please change it.");
+            serviceName = "thorntail/unknown";
```
Comment (jpkrohling) [inline fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java:62]: Can't you derive a better default from the context? For WildFly, we default to the deployment unit's name.

File: fractions/jaeger/src/main/resources/modules/io/jaegertracing/main/module.xml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <module xmlns="urn:jboss:module:1.3" name="io.jaegertracing">
   <resources>
+    <artifact name="io.jaegertracing:jaeger-client:${version.jaeger}"/>
```
Comment (jpkrohling) [inline fractions/jaeger/src/main/resources/modules/io/jaegertracing/main/module.xml]: Do you need this? I think this artifact is empty, apart from a dummy Version class

**Review by jpkrohling — COMMENTED** [review]
Other than the logging of the password, this LGTM. The default service name could be better, but it's not a blocker.

File: fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java (hunk 2 above)
Comment (pavolloffay) [reply fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java:62]: +1, I just don't know how. @Ladicek @kenfinnigan any suggestions?

Comment (jpkrohling) [reply fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java:62]: For reference, this is how it's done in WildFly:
https://github.com/wildfly/wildfly/blob/75584e129564fbd13015f020fa9b9fe72d5e9c79/microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/TracingDeploymentProcessor.java#L114-L134

Comment (kenfinnigan) [reply fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java:62]: I'm not sure if there's a way to get that from inside the deployment itself

Comment (pavolloffay) [reply fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java:62]: Then we cannot get anything better than the current name.

@kenfinnigan anything more to be done before approving/merging?

Comment (pavolloffay) [reply fractions/jaeger/src/main/java/org/wildfly/swarm/jaeger/deployment/JaegerInitializer.java:62]: @kenfinnigan can this be merged?

**Review by Ladicek — APPROVED** [review]

**Comment by Ladicek** [comment]
LGTM. Windows failure (MP Config TCK) is unrelated. Unless @kenfinnigan objects, I'll merge this.

**Pull request merged**